File: tests/fixtures/bench/go/sample.go
// Bench fixture: the golden symbols/edges live in expected.json next to this file.
package sample

import (
	"encoding/json"
	"os"
)

type Store struct {
	Path string
}

func (s *Store) Load() (map[string]any, error) {
	raw, err := os.ReadFile(s.Path)
	if err != nil {
		return nil, err
	}
	var out map[string]any
	err = json.Unmarshal(raw, &out)
	return out, err
}

func ReadConfig(path string) (map[string]any, error) {
	s := &Store{Path: path}
	return s.Load()
}

File: tests/fixtures/bench/javascript/sample.js
// Bench fixture: the golden symbols/edges live in expected.json next to this file.
import { readFile } from "fs/promises";

export class Store {
  constructor(path) {
    this.path = path;
  }

  async load() {
    const raw = await readFile(this.path, "utf8");
    return JSON.parse(raw);
  }
}

export async function readConfig(path) {
  const store = new Store(path);
  return store.load();
}
